packages/pagination/src/index.ts

```diff
@@ -94,201 +109,213 @@ export default defineComponent({
     }
 
-    function getValidCurrentPage(value: number | string) {
-      if (typeof value === 'string') {
-        value = parseInt(value, 10)
-      }
-
-      let resetValue: number | undefined
-
-      if (isNaN(value) || value < 1) {
-        resetValue = 1
-      } else if (internalPageCount.value < value){
-        resetValue = internalPageCount.value
-      }
-
-      return resetValue ?? value
-    }
-
-    watch(() => props.currentPage, val => {
-      internalCurrentPage.value = getValidCurrentPage(val)
-      lastEmittedPage.value = internalCurrentPage.value
-    })
-
-    watch(() => props.pageSize, val => {
-      internalPageSize.value = getValidPageSize(val)
-    })
-
-    watch(
-      () => internalPageCount.value,
-      val => {
-        const oldPage = internalCurrentPage.value
-        if (val > 0 && oldPage === 0) {
-          internalCurrentPage.value = 1
-        } else if (oldPage > val) {
-          internalCurrentPage.value = val === 0 ? 1 : val
-          emitChange()
-        }
-      },
-    )
-
     provide<IPagination>('pagination', {
-      pageCount: computed(() => props.pageCount),
+      pageCount: pageCountBridge,
       disabled: computed(() => props.disabled),
-      currentPage: computed(() => internalCurrentPage.value),
+      currentPage: currentPageBridge,
       changeEvent: handleCurrentChange,
-      handleSizesChange,
+      handleSizeChange,
     })
 
-    return {
-      internalCurrentPage,
-      internalPageSize,
-      lastEmittedPage,
-      userChangePageSize,
-      internalPageCount,
+    return () => {
+      if (!assertValidUsage.value) {
+        warn(componentName, t('el.pagination.docRefer'))
+        return null
+      }
+      if (!props.layout) return null
+      if (props.hideOnSinglePage && pageCountBridge.value <= 1) return null
+      const rootChildren = []
+      const rightWrapperChildren = []
+      const rightWrapperRoot = h('div', { class: 'el-pagination__rightwrapper' }, rightWrapperChildren)
+      const TEMPLATE_MAP = {
+        prev: h(Prev, {
+          disabled: props.disabled,
+          currentPage: currentPageBridge.value,
+          prevText: props.prevText,
+          onClick: prev,
+        }),
+        jumper: h(Jumper),
+        pager: h(Pager, {
+          currentPage: currentPageBridge.value,
+          pageCount: pageCountBridge.value,
+          pagerCount: props.pagerCount,
+          onChange: handleCurrentChange,
+          disabled: props.disabled,
+        }),
+        next: h(Next, {
+          disabled: props.disabled,
+          currentPage: currentPageBridge.value,
+          pageCount: pageCountBridge.value,
+          nextText: props.nextText,
+          onClick: next,
+        }),
+        sizes: h(Sizes, {
+          pageSize: pageSizeBridge.value,
+          pageSizes: props.pageSizes,
+          popperClass: props.popperClass,
+          disabled: props.disabled,
+        }),
+        slot: slots?.default?.() ?? null,
```

Comment:
`Vue` had a public API for rendering slots, you can use it like this:

```ts
import { renderSlots } from 'vue'

{
  // ... other
  slot: renderSlots(slots, 'default', null, () => null),
}
```

Reply:
I double checked with the official documentation site, it is not an official public API but this is how `vue` compiler renders the slots, the way that you used wasn't wrong.

Reply:
kept the original version